### ISSUE TITLE
Ambiguous error message in assert.deepEqual with arrays with custom properties

### ISSUE THREAD

**Issue by lovasoa** [joyent/node#8956]
```javascript
var assert = require("assert");
var m = [1,2];
m.prop = true;
assert.deepEqual([1,2], m);
```

The assertion fails, as expected. But the error message is: `AssertionError: [1,2] deepEqual [1,2]`.
This is not really helpful, and makes debugging harder.

**Comment by cjihrig** [comment]
What would you suggest as a better error message?

**Comment by lovasoa** [comment]
Maybe, if the assertion fails on a custom property in the array, then display the array as an object: `AssertionError: [1,2] deepEqual {0:1, 1:2, m:true}`.

**Comment by cjihrig** [comment]
IMO that message is more confusing. It also really complicates the code, which currently relies on a simple `JSON.stringify()`. I'm -1 on this.

**Comment by jesusabdullah** [comment]
util.inspect is similarly easy to use and should show custom properties:

```
> util.inspect(xs)
'[ 1, 2, foo: \'foo\' ]'
```

**Comment by cjihrig** [comment]
I've already opened #8734 to switch to `util.inspect()` for other reasons, so maybe we can kill two birds with one stone if it lands.

**Comment by cjihrig** [comment]
Closed in bcff90e0c299ce472d3e00b8f886dac8e99478bd
